### PR TITLE
Connection: add IDC secret compare endpoint

### PR DESCRIPTION
**Pull request by sergeymitr**
## Proposed changes:
* Add the "compare secrets" endpoint.
* Always send IDC secret if persistent blog ID is in place.

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
D129990-code

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
The diff does not bring functional changes, so just confirm the connection flow works fine.

### CODE REVIEW

File: projects/packages/identity-crisis/src/class-rest-endpoints.php [hunk 1]
```diff
@@ -219,6 +237,31 @@ public static function fetch_url_secret() {
...
+	 * @return WP_Error|\WP_REST_Response
+	 */
+	public static function compare_url_secret( $request ) {
+		$match = false;
+
+		$storage = new URL_Secret();
+
+		if ( $storage->exists() ) {
+			$remote_secret = $request->get_param( 'secret' );
+			$match         = $remote_secret && $storage->get_secret() === $remote_secret;
```
Comment (anomiex) [inline projects/packages/identity-crisis/src/class-rest-endpoints.php]: Drive-by comment: Should we use [hash_equals](https://www.php.net/hash_equals) here?

Comment (sergeymitr) [reply projects/packages/identity-crisis/src/class-rest-endpoints.php]: Good point, I'll use that 👍

File: projects/packages/identity-crisis/src/class-rest-endpoints.php [hunk 2]
```diff
@@ -233,4 +276,20 @@ public static function url_secret_permission_check() {
...
+
+	/**
+	 * The endpoint is only available on non-connected sites.
+	 * use `/identity-crisis/url-secret` for connected sites.
+	 *
+	 * @return true|WP_Error
+	 */
+	public static function compare_url_secret_permission_check() {
+		return ( new Connection_Manager() )->is_connected()
+			? new WP_Error(
```
Comment (fgiannar) [inline projects/packages/identity-crisis/src/class-rest-endpoints.php:288]: Another drive-by comment: The error code and message here can be misleading, since this is not the result of a permissions' check. Maybe we could replace it with a more relevant `WP_Error`?

File: projects/packages/identity-crisis/src/class-rest-endpoints.php [hunk 3]
```diff
@@ -74,6 +74,24 @@ public static function initialize_rest_api() {
 				'permission_callback' => array( static::class, 'url_secret_permission_check' ),
 			)
 		);
+
+		// Fetch URL verification secret.
+		register_rest_route(
+			'jetpack/v4',
+			'/identity-crisis/compare-url-secret',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
```
Comment (fgiannar) [inline projects/packages/identity-crisis/src/class-rest-endpoints.php:83]: Would it maybe make more sense for this to be a `GET` request? We are not modifying anything, right? Just comparing the secrets.

Comment (sergeymitr) [reply projects/packages/identity-crisis/src/class-rest-endpoints.php:83]: Hi @fgiannar, thanks for taking a look!
The idea here is to keep the secret from showing up in the access log, which will happen if we use GET.
So it may be not be quite REST-ful, but safer that way 🙂

File: projects/packages/identity-crisis/src/class-rest-endpoints.php (hunk 2 above)
Comment (sergeymitr) [reply projects/packages/identity-crisis/src/class-rest-endpoints.php:288]: Good point, fixed 👍

**Review by bindlegirl — APPROVED** [review]
As far as I can tell, this works
